app/api/events.py

```diff
@@ -550,7 +556,12 @@ def before_update_object(self, event, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
-        if data.get('starts_at') != event.starts_at or data.get('ends_at') != event.ends_at:
+        # 1 - call validate_date if either event start_at or end_at is updated
+        # 2 - call validate_date if drafted event is being published
+        # 3 - call validate_date if event is being restored
+        if (data.get('starts_at') != event.starts_at or data.get('ends_at') != event.ends_at) or \
+                (event.state == "draft" and data.get('state') == "published") or \
+                (event.deleted_at and not data.get('deleted_at')):
```

Comment:
Still one big complex condition. Split in 3 booleans as mentioned previously. This is the third time now

Reply:
Maybe I'm not getting you.
Do you want me to have 3 different functions that return true or false and each function have one of the conditions ?

Reply:
a = blah
b = blah
c =blah
a or b or c